test/externalTests/gnosis.sh

```diff
@@ -87,33 +87,39 @@ function gnosis_safe_test
     sed -i "s|it\(('can only be called from Safe itself'\)|it.skip\1|g" test/libraries/Migration.spec.ts
     sed -i "s|it\(('should enforce delegatecall to MultiSend'\)|it.skip\1|g" test/libraries/MultiSend.spec.ts
 
+    jq '.overrides."@ethersproject/abstract-provider"="5.6.0" |
+        .overrides."@ethersproject/abstract-signer@5.6.0"
+                  ."@ethersproject/abstract-provider"="5.6.0"' package.json > package.json.tmp
+    mv package.json.tmp package.json
+
     neutralize_package_lock
     neutralize_package_json_hooks
     force_hardhat_compiler_binary "$config_file" "$BINARY_TYPE" "$BINARY_PATH"
     force_hardhat_compiler_settings "$config_file" "$(first_word "$SELECTED_PRESETS")" "$config_var"
-    npm install
-    npm install hardhat-gas-reporter
+    npm install npm@8.11.0
```

Comment:
We don't need a specific version. We want the latest one.
```suggestion
    # We need the latest npm version for the `overrides` feature in package.json
    npm install npm
```

Or, if you're worried about the `overrides` feature:
```suggestion
    # We need an npm version that supports the `overrides` key in package.json
    npm install npm@>=8.3.0
```